File: src/main/java/org/ecews/renaissance/domain/Triage.java
/*
* MIT License
*
* Copyright (c) 2025 [AUTHOR OR ORGANIZATION]
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in
* all copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
* THE SOFTWARE.
*/
package org.ecews.renaissance.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "ren_triage")
public class Triage extends AuditEntity {
    
    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    
    @ManyToOne(optional = false)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;
    
    // Medical History
    private Boolean diabetes;
    private Boolean asthma;
    private Boolean sickleCell;
    private Boolean smoking;
    
    // Anthropometry
    private Double weight;
    private Double height;
    
    // Vital Signs
    private Double temperature;
    @Column(columnDefinition = "systolic_bp")
    private Integer systolicBp;
    @Column(columnDefinition = "diastolic_bp")
    private Integer diastolicBp;
    private Integer pulseRate;
    private Integer respiratoryRate;
}

File: Renaissance.AspNet/backend/Renaissance.Domain/Entities/Triage.cs
using System.Text.Json.Serialization;

namespace Renaissance.Domain.Entities;

public class Triage : AuditEntity
{
    public Guid Id { get; set; }
    public Guid PatientId { get; set; }

    [JsonIgnore]
    public Patient? Patient { get; set; }

    public bool? Diabetes { get; set; }
    public bool? Asthma { get; set; }
    public bool? SickleCell { get; set; }
    public bool? Smoking { get; set; }
    public double? Weight { get; set; }
    public double? Height { get; set; }
    public double? Temperature { get; set; }
    public int? SystolicBp { get; set; }
    public int? DiastolicBp { get; set; }
    public int? PulseRate { get; set; }
    public int? RespiratoryRate { get; set; }
}
